postproengine/wake_meandering.py

```diff
@@ -0,0 +1,406 @@
+# Get the location where this script is being run
+import sys, os
+scriptpath = os.path.dirname(os.path.realpath(__file__))
+basepath   = os.path.dirname(scriptpath)
+# Add any possible locations of amr-wind-frontend here
+amrwindfedirs = ['../',
+                 basepath]
+for x in amrwindfedirs: sys.path.insert(1, x)
+
+from postproengine import registerplugin, mergedicts, registeraction
+import postproamrwindsample_xarray as ppsamplexr
+import postproamrwindsample as ppsample
+import numpy as np
+import pandas as pd
+import os
+import pickle
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import importlib.util
+import sys
+
+# Add check for samwich package
+name = 'samwich'
+if (spec := importlib.util.find_spec(name)) is not None:
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[name] = module
+    spec.loader.exec_module(module)
+    from samwich.dataloaders import PlanarData
+    from samwich.waketrackers import track
+    usesamwich=True
+else:
+    print(f"can't find the {name!r} module")
+    usesamwich=False
+
+"""
+Plugin for computing wake meandering statistics
+
+See README.md for details on the structure of classes here
+"""
+
+def get_wake_centers(u,YY,ZZ,method='ConstantArea',weighting=lambda u: np.ones_like(u),args=None):
+    datadict = {}
+    datadict['y'] = np.copy(YY[:,:].T)
+    y_grid_center = (datadict['y'][-1,0] + datadict['y'][0,0])/2.0 
+    datadict['z'] = np.copy(ZZ[:,:].T)
+    datadict['u'] = u[:,:,:,0].transpose(0,2,1).copy()
```

Comment:
need check that data is ordered (y,z). (z,y) will not work. This depends on how it is stored in the AMR-Wind netcdf file